accepted/future-releases/class-modifiers/feature-specification.md

```diff
@@ -687,6 +687,26 @@ It's a compile-time error if:
     class M with S {} // Error, for several reasons.
     ```
 
+*   A mixin has a declared `on` type from another library which is marked
```

Comment:
If we do need an extra case for `final`, I'd go all in and declare all the errors here.

> *   A declaration has a direct super declaration from another library, 
>      which is marked `final`.
>
>      More formally:
>      A declaration *D* from library *L* has a direct superdeclaration *S* 
>      marked `final` (so necessarily a `class` declaration) in library *K*, and neither
>      * *L* and *K* is the same library, nor
>      * *K* is a platform library and *L* is a pre-feature library.
> 
>      ```dart
>     // a.dart 
>     final class F {}
>
>     // b.dart    
>     import 'a.dart';
>     
>     class C1 extends F {}  // Error.
>     class C2 implements F {}  // Error.
>     mixin class C3 implements F {}  // Error.
>     mixin M1 implements F {}  // Error.
>     enum E1 implements F {}  // Error.
> ```

Then remove `final` from the direct `interface` superdeclaration cases below.

I think it could be cleaner that way.

(The transitive `base`/`final` restriction case is still needed, although even there I think the `final` wouldn't be needed, since any transitive chain to a final class in another library, which doesn't have an otherwise invalid direct step, will also contain at least one `base` declaration in the same library as the `final` library.)

Reply:
Yes, that's cleaner. Did that.